### ISSUE TITLE
feat(ml): migrate to PyTorch for deep learning capabilities and Fabric integration

### ISSUE THREAD

**Issue by ibuyspy** [IBuySpy-Dev/AwardPredictor#89]
## Summary
Migrate the ML stack from scikit-learn + XGBoost to PyTorch for better deep learning capabilities, distributed training support, and production-grade model inference in Fabric.

## Current State
- Model: `src/models/train.py` uses XGBoost (shallow tree-based model)
- Training: sklearn Pipeline with SimpleImputer + XGBClassifier
- Evaluation: ROC-AUC on test set only
- Inference: Basic predict_proba for win probabilities

## Target State
- Model: PyTorch neural network with appropriate architecture for tabular award prediction
- Training: PyTorch Lightning for distributed training, logging, checkpointing
- Evaluation: Multiple metrics (AUC, precision, recall, F1) with cross-validation
- Inference: TorchScript export for production deployment to Fabric
- Reproducibility: Seed management, deterministic ops, model versioning

## Why PyTorch
1. **Deep Learning**: Capture non-linear relationships in award/artist features
2. **Production-Ready**: TorchScript export, ONNX support for Fabric deployment
3. **Distributed**: Multi-GPU training on Fabric Spark clusters via PyTorch Distributed
4. **Monitoring**: Built-in logging, early stopping, LR scheduling
5. **Enterprise**: Industry standard for ML ops, extensive documentation

### 1. Define PyTorch Model Architecture
- Input: Tabular features (normalized, imputed)
- Layers: Embeddings for categorical fields (domain, year, category), dense layers for continuous features
- Output: Sigmoid for binary classification (winner/non-winner)
- Reference: TabNet, VIME, or custom MLP

### 2. Create `src/models/pytorch_model.py`
- Model class inheriting `torch.nn.Module`
- Forward pass with feature embedding and dense layers
- Handle missing values (or require pre-imputation from pipeline)
- Save/load via `torch.save()` / `torch.load()`

### 3. Update `src/models/train.py`
- Replace XGBoost with PyTorch Lightning Trainer
- Use PyTorch Lightning datamodules for train/test splits
- Add cross-validation with fold averaging
- Log metrics to MLflow (for Fabric integration)
- Export model to TorchScript (`.pt` format)

### 4. Update `src/models/predict.py`
- Load TorchScript model (no Python pickle needed)
- Batch inference with torch.no_grad()
- Output predictions as DataFrame (same schema as before)

### 5. Dependencies
Add to `requirements.txt`:
```
torch>=2.0
pytorch-lightning>=2.0
torchvision
```

### 6. Tests
- `tests/test_pytorch_model.py` — Model forward pass, output shape
- `tests/test_pytorch_training.py` — Training loop, checkpointing, metrics
- Ensure XGBoost tests still pass during transition (run both models in parallel)

### 7. Deployment
- Save trained model as `.pt` (TorchScript)
- Fabric notebook loads via `torch.load()` and runs inference
- Document model serving in docs/DEPLOYMENT.md

## Migration Path
1. Implement PyTorch model in parallel (don't remove XGBoost yet)
2. Compare performance (AUC, inference latency) on same train/test split
3. Switch to PyTorch in `src/models/train.py` if better results
4. Deprecate XGBoost in future PR
5. Update pipeline orchestrator to use new model

## References
- [PyTorch Docs](https://pytorch.org/docs/stable/index.html)
- [PyTorch Lightning](https://lightning.ai/)
- [Tabular Deep Learning](https://arxiv.org/abs/2207.08815)
- [TorchScript Export](https://pytorch.org/docs/stable/jit.html)

## Dependencies
- #84 (Fabric repo structure — run on Fabric Spark)
- None blocking (can implement in parallel with current XGBoost)
